internal/tasks/resourcesync.go

```diff
@@ -367,3 +376,27 @@ func isValidFile(filename string) bool {
 	}
 	return false
 }
+
+func (r *ResourceSync) validateFleetNameConflicts(ctx context.Context, fleets []*api.Fleet, owner string) error {
+	var conflictingFleets []string
+
+	for _, fleet := range fleets {
+		// Check if a fleet with this name already exists
+		existingFleet, status := r.serviceHandler.GetFleet(ctx, *fleet.Metadata.Name, api.GetFleetParams{})
```

Comment:
Ideally this check would be in the store layer to make sure nobody is modifying a resource owned by someone else, but this is OK for now.